### PR TITLE
add com.nickgirga.chocolate-box

### PR DESCRIPTION
**Pull request by nickgirga**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

My app requires the `--talk-name=org.freedesktop.Flatpak` permission, as it needs to use `flatpak-spawn` to run other programs that are installed on the system. My app is a DOOM launcher that makes use of other DOOM source ports installed on the system (e.g. Chocolate DOOM, Crispy DOOM, GZDoom, etc). These source ports should not be bundled in the flatpak of my app because the user may only want to use it with a single source port. The user may want to use my app to launch a source port that has been installed on the system itself, rather than another flatpak as well. This is an integral feature to being able to launch pretty much anything in my app. I am open to using other methods to achieve this, but I couldn't find much documentation of working around the problem that my app faces when in a flatpak sandbox.

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: com.nickgirga.chocolate-box.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - install -D res/icon/icon_128.png /app/share/icons/chocolate-box_128.png
+      - install -D res/icon/icon_256.png /app/share/icons/chocolate-box_256.png
+      - install -D res/icon/com.nickgirga.chocolate-box.png /app/share/icons/hicolor/256x256/apps/com.nickgirga.chocolate-box.png
+      - install -D com.nickgirga.chocolate-box.desktop /app/share/applications/com.nickgirga.chocolate-box.desktop
+      - install -Dm644 com.nickgirga.chocolate-box.metainfo.xml -t /app/share/metainfo
+
+    sources:
+      - type:   archive
+        url:    https://gitlab.com/nickgirga/chocolate-box/-/archive/1.4/chocolate-box-1.4.tar.gz
+        sha256: a3a4102421fd14d1fe8a12f3644cf306b844904dd29f2e5b0339a4fba9e4953e
```
Comment (bbhtt) [inline com.nickgirga.chocolate-box.yml:38]:
```suggestion
      - type: archive
        url: https://gitlab.com/nickgirga/chocolate-box/-/archive/1.4/chocolate-box-1.4.tar.gz
        sha256: a3a4102421fd14d1fe8a12f3644cf306b844904dd29f2e5b0339a4fba9e4953e
```

File: com.nickgirga.chocolate-box.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+base-version: stable-23.08
+
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
+  # access to org.freedesktop.Flatpak; for flatpak Source port launching
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (bbhtt) [inline com.nickgirga.chocolate-box.yml:17]: Needs an exception in the linter https://docs.flathub.org/docs/for-app-authors/linter#exceptions

File: com.nickgirga.chocolate-box.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  # GPU acceleration if needed
+  - --device=dri
+  # access to home files; for locating WADs and custom installed source ports
+  - --filesystem=home
```
Comment (hfiguiere) [inline com.nickgirga.chocolate-box.yml:21]: what the point of this you punch the other hole to run things out of the sandbox?

File: com.nickgirga.chocolate-box.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+id: com.nickgirga.chocolate-box
+runtime: org.gnome.Platform
+runtime-version: '45'
+sdk: org.gnome.Sdk
+command: chocolate-box
+
+base: org.winehq.Wine
```
Comment (hfiguiere) [inline com.nickgirga.chocolate-box.yml:7]: why do you need this?

**Pull request closed without merging**